axelrod/strategies/darwin.py

```diff
@@ -0,0 +1,63 @@
+import inspect
+from axelrod import Player
+
+
+class Darwin(Player):
+    """ A strategy which accumulates a record (the 'genome') of what the most
+        favourable response in the previous round should have been, and naively
+        assumes that this will remain the correct response at the same round of
+        future trials.  This 'genome' is preserved between opponents.
+        If no record yet exists, the opponent's response from the previous round
+        is returned.
+    """
+
+    name = "Darwin"
+    genome = ['C']
+    valid_callers = ["play"]    # What functions may invoke our strategy.
+    outcomes = { ('C','C') : 1,
+                 ('C','D') : -5,
+                 ('D','C') : 5,
+                 ('D','D') : -1
+               }
+
+    def __init__(self):
+        super(self.__class__, self).__init__()
+        self.response = self.__class__.genome[0]
+
+
+    def strategy(self, opponent):
+        # Frustrate psychics and ensure that simulated rounds
+        # do not influence genome.
+        if inspect.stack()[1][3] not in self.__class__.valid_callers:
+            return 'C'
+
+        trial = len(self.history)
+
+        if trial > 0:
+            outcome = self.__class__.outcomes[(self.history[-1], opponent.history[-1])]
+            self.mutate(outcome, trial)
+            # Update genome with selected response
+            self.__class__.genome[trial-1] = self.response
+
+        if trial < len(self.__class__.genome):
+            # Return response from genome where available...
+            current = self.__class__.genome[trial]
+        else:
+            # ...otherwise use Tit-for-Tat
+            self.__class__.genome.append(opponent.history[-1])
+            current = opponent.history[-1]
+
+        return current
+
+
+    def reset(self):
+        """ Reset instance properties. """
+        self.history = []
+        self.__class__.genome[0] = 'C' # Ensure initial Cooperate
```

Comment:
I'm afraid that this is definitely a 'cheating' per say strategy. Some of the q-learners wanted to do this sort of thing.